src/js/components/toolbar/MarkDropPlaceGroup.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {State} from '../../store';
+import {PlaceHolderRecord} from '../../store/factory/Layout';
+import MarkDropPlace from './MarkDropPlace';
+
+interface StateProps {
+  layoutId: number;
+  placeholders: PlaceHolderRecord[];
+}
+
+function mapStateToProps(state: State): StateProps {
+  const layouts = state.getIn(['vis', 'present', 'layouts']);
+
+  const layoutId = Array.from(layouts.keys())[0];
+  // const layout = layouts.filter(layout => layout._id == ind);
+  const placeholders = state.getIn(['vis', 'present', 'layouts', layoutId, 'placeHolders']);
+
+  return {
+    layoutId: +layoutId,
```

Comment:
why cast to number? if you're using the `layoutId` to look up entries in the state, which is a `Map<string, LayoutRecord>`, it will need to be a string anyway